### ISSUE TITLE
User account can be approved before email address is confirmed

### ISSUE THREAD

**Issue by patrickakk** [acdh-oeaw/dhcr-main#108]
Problem:
User account can be approved before email address is confirmed

Solution:

- [x] a) Update UsersController starting at line 490 (after migration to new framework version is finished)
- [x] b) Test if user can login if email is not confirmed, else set redirect

@IvdL22 @PixlTracer 
FYI

**Comment by patrickakk** [comment]
a) Implemented
b) This works correct. The user is redirected and can request the confirmation mail again.

**Comment by patrickakk** [comment]
@IvdL22 @PixlTracer 

I've solved it and tested it. It works. Now you can only see (and approve) a new user, after they confirmed their email address.

Do you want to review this issue?
(The implementation is currently only available in dev)

**Comment by PixlTracer** [comment]
thank you @patrickakk 
if there a no objections, I'll set the label to done; in case we encounter difficulties in production, we might get back to this issue

**Comment by patrickakk** [comment]
@PixlTracer Thank you for your cooperation. Although

> in case we encounter difficulties in production, we might get back to this issue

Sounds similar to "we'll test it in production" ? Maybe that might not be the preferred approach? There is an test instance available especially for this. This resource is specially available to try out and test if necessary. So that, when it is approved and goes into production, then, preferably, no changes should be needed. Since it has been tested before ;)

The reason that I asked "Do you want to review this issue?" was that it might be to technical for you to review it?

Currently I've marked the issue as blocked. My suggestion would be to wait until it's available in test and then you can try out everything you want.

Would that work for you as well?

**Comment by patrickakk** [comment]
@IvdL22 @PixlTracer 

Can you review this?

**Comment by PixlTracer** [comment]
hi @patrickakk 
I've tested this in dev;

kudos to you, for this very nice info-message!
<img width="1069" alt="Bildschirmfoto 2024-11-11 um 11 56 48" src="https://github.com/user-attachments/assets/395237f2-1722-4f05-acc3-78bf5db8511a">

I've logged on again and couldn't see my private email which is good, because I (as the one who wants to register) did not approve it yet. 
after approving my private address, I got this screen (screenshotted from mobile), which is also really nice, well done!!
![WhatsApp Image 2024-11-11 at 12 03 30](https://github.com/user-attachments/assets/680e4745-99fd-4600-9f65-58f8e877e6c7)

then, I checked again the dashboard (as admin) - and I saw the private email indicated.

so this works fine, thank you for implementation!
I'll set the label to done.
